gobblin-core/src/main/java/gobblin/writer/FsDataWriter.java

```diff
@@ -49,17 +56,35 @@ public FsDataWriter(State properties, String fileName, int numBranches, int bran
     // Add all job configuration properties so they are picked up by Hadoop
     JobConfigurationUtils.putStateIntoConfiguration(properties, conf);
 
-    // Initialize file system
-    String uri = properties.getProp(
-        ForkOperatorUtils.getPropertyNameForBranch(ConfigurationKeys.WRITER_FILE_SYSTEM_URI, numBranches, branchId),
-        ConfigurationKeys.LOCAL_FS_URI);
-    this.fs = FileSystem.get(URI.create(uri), conf);
+    String uri =
+        properties
+            .getProp(ForkOperatorUtils.getPropertyNameForBranch(ConfigurationKeys.WRITER_FILE_SYSTEM_URI, numBranches,
+                branchId), ConfigurationKeys.LOCAL_FS_URI);
+
+    if (properties.getPropAsBoolean(ConfigurationKeys.SHOULD_FS_PROXY_AS_USER,
+        ConfigurationKeys.DEFAULT_SHOULD_FS_PROXY_AS_USER)) {
+      // Initialize file system as a proxy user.
+      try {
+        this.fs =
+            new ProxiedFileSystemWrapper().getProxiedFileSystem(properties, ProxiedFileSystemWrapper.AuthType.TOKEN,
+                properties.getProp(ConfigurationKeys.FS_PROXY_AS_USER_TOKEN_FILE), uri);
+      } catch (InterruptedException e) {
+        Throwables.propagate(e);
```

Comment:
Instead of propagating the exceptions, I suggest you simply wraps them into an `IOException` and throw that instead.